### PR TITLE
[COMCTL32] Fix desktop arrow keys

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

Fix the behaviors of keyboard arrow keys on desktop.

JIRA issue: [CORE-16875](https://jira.reactos.org/browse/CORE-16875)

## Proposed changes

- Disable special auto-arrange codes in `LISTVIEW_GetNextItem` function.
- Improve `LISTVIEW_SetItemPosition` function for auto-arrange.

BEFORE:
before
Arrow keys won't work correctly.

AFTER:
after
Arrow keys works well.

### CODE REVIEW

**Comment by HBelusca** [comment]
But does it still perform auto-arrange functionality if you move the icons in different places and then select the option in the explorer desktop context menu?

**Comment by katahiromz** [comment]
Auto-arrange is still not implemented correctly.

**Comment by KyleKatarn31** [comment]
Does this needs a WineHQ patch too ?

**Comment by katahiromz** [comment]
Now we can change the ordering by dragging & dropping.
can-change-order2
can-change-order1

**Comment by katahiromz** [comment]
The DPAs have to be sorted when it is assumed as sorted and it has been modified.

File: dll/win32/comctl32/listview.c [hunk 1]
```diff
@@ -9044,6 +9054,109 @@ static BOOL LISTVIEW_SetItemCount(LISTVIEW_INFO *infoPtr, INT nItems, DWORD dwFl
...
+        ranges_additem(infoPtr->selectionRanges, i2);
+    }
+    else if (!b1 && b2)
+    {
+        ranges_additem(infoPtr->selectionRanges, i1);
+        ranges_delitem(infoPtr->selectionRanges, i2);
+    }
+}
+
+static void SwapRelay(LISTVIEW_INFO *infoPtr, INT iFrom, INT iTo)
```
Comment (HBelusca) [inline dll/win32/comctl32/listview.c:9098]: What does this function do exactly? (what does "relay" mean in this context?)

**Comment by katahiromz** [comment]
Superseded by #3162.

**Pull request closed without merging**